src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -451,3 +468,41 @@ BuildDistinctPlan(Query *masterQuery, Plan *subPlan)
 
 	return distinctPlan;
 }
+
+
+/*
+ * IsGroupBySubsetOfDistinct checks whether each clause in group clauses also
+ * exists in the distinct clauses.
+ */
+static bool
+IsGroupBySubsetOfDistinct(Query *masterQuery)
+{
```

Comment:
I think this function should return `false` if there is no `GROUP BY`. That shouldn't affect us here?